openwisp_notifications/tests/test_notifications.py

```diff
@@ -944,6 +944,46 @@ def test_notification_for_unverified_email(self):
         # we don't send emails to unverified email addresses
         self.assertEqual(len(mail.outbox), 0)
 
+    @patch('openwisp_notifications.tasks.batch_email_notification.apply_async')
+    def test_batch_email_notification_with_descriptions(self, mock_send_email):
+        for _ in range(5):
+            notify.send(recipient=self.admin, **self.notification_options)
+
+        # Check if only one mail is sent initially
+        self.assertEqual(len(mail.outbox), 1)
+
+        # Call the task
+        tasks.batch_email_notification(self.admin.id)
+
+        # Check if the rest of the notifications are sent in a batch
+        self.assertEqual(len(mail.outbox), 2)
+        self.assertIn('Summary of 4 Notifications', mail.outbox[1].subject)
+        self.assertNotIn('View all Notifications', mail.outbox[1].body)
+        self.assertIn('Test Notification', mail.outbox[1].body)
+
+    @patch('openwisp_notifications.tasks.batch_email_notification.apply_async')
+    def test_batch_email_notification_with_call_to_action(self, mock_send_email):
+        self.notification_options.update(
+            {
+                'message': 'Notification title',
+                'type': 'default',
+            }
+        )
+        for _ in range(11):
+            notify.send(recipient=self.admin, **self.notification_options)
+
+        # Check if only one mail is sent initially
+        self.assertEqual(len(mail.outbox), 1)
+
+        # Call the task
+        tasks.batch_email_notification(self.admin.id)
+
+        # Check if the rest of the notifications are sent in a batch
+        self.assertEqual(len(mail.outbox), 2)
+        self.assertIn('Summary of 10 Notifications', mail.outbox[1].subject)
+        self.assertIn('View all Notifications', mail.outbox[1].body)
+        self.assertNotIn('Test Notification', mail.outbox[1].body)
+
```

Comment:
Can we add another test in which this feature is disabled and ensure that no batching happens in that case?
Using the `override_settings` from django or mocking the setting should work.